src/pymor/algorithms/gram_schmidt.py

```diff
@@ -84,6 +84,8 @@ def gram_schmidt(A, product=None, return_R=False, atol=1e-13, rtol=1e-13, offset
                         continue
                     p = A[j].pairwise_inner(A[i], product)[0]
                     A[i].axpy(-p, A[j])
+                    common_dtype = np.promote_types(R.dtype, p.dtype)
```

Comment:
I guess `p` is a scalar, so it doesn't have `dtype` attribute.
```suggestion
                    common_dtype = np.promote_types(R.dtype, type(p))
```